pkg/apis/core/validation/cloudprofile.go

```diff
@@ -309,21 +305,12 @@ func validateContainerRuntimesInterfaces(cris []core.CRI, fldPath *field.Path) f
 		}
 		duplicateCRI.Insert(string(cri.Name))
 
-		if cri.Name == core.CRINameDocker {
-			hasDocker = true
-		}
-
-		if !availableWorkerCRINamesForCloudProfile.Has(string(cri.Name)) {
-			allErrs = append(allErrs, field.NotSupported(criPath, cri, sets.List(availableWorkerCRINamesForCloudProfile)))
+		if !availableWorkerCRINames.Has(string(cri.Name)) {
+			allErrs = append(allErrs, field.NotSupported(criPath, cri, sets.List(availableWorkerCRINames)))
```

Comment:
Not introduced with this PR but:
```suggestion
			allErrs = append(allErrs, field.NotSupported(criPath, string(cri.Name), sets.List(availableWorkerCRINames)))
```

in order to have:
```
The CloudProfile "local" is invalid: spec.machineImages[0].versions[0].cri[0]: Unsupported value: "containerd1": supported values: "containerd"
```

instead of 
```
The CloudProfile "local" is invalid: spec.machineImages[0].versions[0].cri[0]: Unsupported value: core.CRI{Name:"containerd1", ContainerRuntimes:[]core.ContainerRuntime(nil)}: supported values: "containerd"
```